sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -52,5 +59,74 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
             Policies ??= new();
             Policies.Add((position, policy));
         }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="statusCodes">Status codes that will not be considered to be error status codes for the scope of the service method call.</param>
+        /// <param name="isError">Whether the passed-in status codes will be considered to be error codes for the duration of this request.</param>
+        public void ConfigureResponse(int[] statusCodes, bool isError)
+        {
+            if (isError)
+            {
+                _customErrorCodes ??= new();
+                _customErrorCodes.AddRange(statusCodes);
+            }
+            else
+            {
+                _customNonErrorCodes ??= new();
+                _customNonErrorCodes.AddRange(statusCodes);
+            }
+        }
+
+        internal ResponseClassifier GetResponseClassifier(ResponseClassifier inner)
```

Comment:
Is there a way to avoid response classifier wrapping? i.e. have just one classifier object? I honestly dont know as I forgot how the inner classifier works.

Reply:
Is the concern allocations?  

The inner response classifier could have been set by the client author, so we don't know what logic it's using.

Reply:
The new approach addresses this -- the design goal is to have all of the classifier logic in a single place to it is easy to see at a glance what the classifier is doing.